include/bbp/sonata/config.h

```diff
@@ -76,6 +76,8 @@ struct SONATA_API PopulationProperties {
 class SONATA_API CircuitConfig
 {
   public:
+    enum class Type { invalid, Complete, Partial };
```

Comment:
You could reuse some of the MR documentation to explain what complete and partial mean:

> * a partial circuit config has a `metadata` property `type` that is equal to `partial`
> * this relaxes some checking that is done on circuit configs:
>   * can be missing the top level `networks` key
>   * can be missing the `nodes` and `edges` properties under `network`
> * mandatory properties aren't enforced (for example biophysical
>   properties don't need a `[alternative_]morphologies_dir`

Reply:
agreed.

Reply:
Why is `invalid` lowercase, when all others are capitalized? This somehow really bugs me.

Reply:
> Why is invalid lowercase, when all others are capitalized?

See https://github.com/BlueBrain/libsonata/pull/241#discussion_r1010176908

>  This somehow really bugs me.

Same.

Reply:
Ah, missed the spec part :(

Reply:
> You could reuse some of the MR documentation to explain what complete and partial mean:

Agreed, will add doc.